File: src/main/java/egovframework/com/cmm/filter/PathRewriteFilter.java
package egovframework.com.cmm.filter;

import javax.servlet.*;
import javax.servlet.http.HttpServletRequest;
import java.io.IOException;

public class PathRewriteFilter implements Filter {
    @Override
    public void init(FilterConfig filterConfig) throws ServletException {}

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        HttpServletRequest req = (HttpServletRequest) request;
        String contextPath = req.getContextPath();
        String uri = req.getRequestURI();
        String prefix = contextPath + "/pst_webapp";
        if (uri.startsWith(prefix)) {
            String forwardTo = uri.substring(prefix.length());
            if (forwardTo.isEmpty()) forwardTo = "/";
            RequestDispatcher rd = request.getRequestDispatcher(forwardTo);
            rd.forward(request, response);
            return;
        }
        chain.doFilter(request, response);
    }

    @Override
    public void destroy() {}
}

File: src/main/Kotlin/egovframework/com/cmm/filter/PathRewriteFilter.kt
package egovframework.com.cmm.filter

import java.io.IOException
import javax.servlet.*
import javax.servlet.http.HttpServletRequest

class PathRewriteFilter : Filter {
    @Throws(ServletException::class)
    override fun init(filterConfig: FilterConfig?) {
    }

    @Throws(IOException::class, ServletException::class)
    override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain) {
        val req = request as HttpServletRequest
        val contextPath = req.getContextPath()
        val uri = req.getRequestURI()
        val prefix = contextPath + "/pst_webapp"
        if (uri.startsWith(prefix)) {
            var forwardTo = uri.substring(prefix.length)
            if (forwardTo.isEmpty()) forwardTo = "/"
            val rd = request.getRequestDispatcher(forwardTo)
            rd.forward(request, response)
            return
        }
        chain.doFilter(request, response)
    }

    override fun destroy() {}
}

